File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.py
import sys

T = int(sys.stdin.readline().strip())

# fb(0) ~ fb(1)까지의 0의 개수
zeros = [1, 0]

# fb(0) ~ fb(1)까지의 1의 개수
ones = [0, 1]

# N이 40보다 작거나 같은 자연수이므로 fb(40)까지의 값을 저장
for i in range(2, 41):
    zeros.append(zeros[i-1] + zeros[i-2])
    ones.append(ones[i-1] + ones[i-2]) 

for _ in range(T):
    n = int(sys.stdin.readline().strip())
    print(zeros[n], end=' ')
    print(ones[n])

File: 백준/Silver/1003. 피보나치 함수/피보나치 함수.java
import java.io.*;

// https://www.acmicpc.net/problem/1003
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int[] zeros = new int[41];
        zeros[0] = 1;
        zeros[1] = 0;

        int[] ones = new int[41];
        ones[0] = 0;
        ones[1] = 1;

        for(int i = 2; i <= 40; i++){
            zeros[i] = zeros[i - 1] + zeros[i - 2];
            ones[i] = ones[i - 1] + ones[i - 2];
        }

        StringBuilder sb = new StringBuilder();

        int T = Integer.parseInt(br.readLine());
        for (int t = 0; t < T; t++) {
            int N = Integer.parseInt(br.readLine());
            sb.append(zeros[N]).append(" ").append(ones[N]).append("\n");
        }
        System.out.println(sb);
    }
}